File: TennisObjects/Classification.cs
using System;
using System.Collections.Generic;
using System.Text;
using Csla;
using log4net;
using Bahns.Data;

namespace TennisObjects
{
	[Serializable()]
	public class Classification : BusinessBase<Classification>
	{

		private static ILog log = LogManager.GetLogger(typeof(Classification).ToString());

		#region Private Data
		private Int32 _ID;
		private string _Name;
		private string _USTAEquivalent;
		#endregion

		#region Validation
		protected override void AddBusinessRules()
		{
			//BrokenRules.Assert("NameReq", "Name is required", _Name.Length == 0);

		}

		protected override void AddInstanceBusinessRules()
		{

		}
		#endregion

		#region Properties
		public Int32 ID
		{
			get { return _ID; }
		}

		public string Name
		{
			get { return _Name; }
			set { SetProperty(ref _Name, value); }
		}

		public string USTAEquivalent
		{
			get { return _USTAEquivalent; }
			set { SetProperty(ref _USTAEquivalent, value); }
		}
		#endregion

		#region System.Object Overrides
		public override string ToString()
		{
			return Name;
		}

		protected override object GetIdValue()
		{
			return _ID;
		}
		#endregion

		#region Shared Methods
		public static Classification NewClassification()
		{
			return new Classification();
		}

		public static Classification NewClassification(string Name, string USTAEquivalent)
		{
			return new Classification(Name, USTAEquivalent);
		}

		public static Classification GetClassification(SafeDataReader dr)
		{
			Classification Classification = new Classification();
			Classification.Fetch(dr);
			return Classification;
		}
		#endregion

		#region Constructors
		private Classification(string Name, string USTAEquivalent)
		{
			MarkAsChild();
			_Name = Name;
			_USTAEquivalent = USTAEquivalent;
		}

		private Classification()
		{
			//Prevent direction creation 
			MarkAsChild();
		}
		#endregion

		#region Data Access
		private void Fetch(SafeDataReader dr)
		{
			_ID = dr.GetInt32();
			_Name = dr.GetString();
			_USTAEquivalent = dr.GetString();
			MarkOld();
		}

		internal void Update()
		{
			if (this.IsDeleted)
			{
				if (!this.IsNew)
				{
					DbHelper.ExecuteNonQuery("csla_delete_classification", Param.In("@ID", _ID));
					//DbHelper.ExecuteNonQuery("csla_delete_classification", ID);
				}
				MarkNew();
			}
			else
			{
				DbHelper.ExecuteNonQuery("csla_save_classification", Param.ID("@ID", _ID, IsNew), Param.In("", _Name), Param.In("", _USTAEquivalent));
				//DbHelper.ExecuteNonQuery("csla_save_classification", Util.IDParam(_ID, IsNew), _Name, _USTAEquivalent);
				MarkOld();
			}
		}
		#endregion

	}

}

File: Objects/Classification.vb
<Serializable()> _
Public Class Classification
    Inherits BusinessBase

    Private Shared log As ILog = LogManager.GetLogger(GetType(Classification).ToString())

#Region "Private Data"
    Private _ID As Int32
    Private _Name As String
    Private _USTAEquivalent As String
#End Region

#Region "Business Properties and Methods"
    Public ReadOnly Property ID() As Int32
        Get
            Return _ID
        End Get
    End Property

    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
            MarkDirty()
            BrokenRules.Assert("NameReq", "Name is required", _Name.Length = 0)
        End Set
    End Property

    Public Property USTAEquivalent() As String
        Get
            Return _USTAEquivalent
        End Get
        Set(ByVal value As String)
            _USTAEquivalent = value
            MarkDirty()
        End Set
    End Property
#End Region

#Region "System.Object Overrides"
    Public Overrides Function ToString() As String
        Return Name
    End Function

    Public Overloads Function Equals(ByVal A As Classification) As Boolean
        Return _ID = A._ID
    End Function

    Public Overrides Function GetHashCode() As Integer
        Return _ID.GetHashCode()
    End Function
#End Region

#Region "Shared Methods"
    Public Shared Function NewClassification() As Classification
        Return New Classification()
    End Function

    Public Shared Function NewClassification(ByVal Name As String, ByVal USTAEquivalent As String) As Classification
        Return New Classification(Name, USTAEquivalent)
    End Function

    Public Shared Function GetClassification(ByVal dr As SafeDataReader) As Classification
        Dim Classification As New Classification
        Classification.Fetch(dr)
        Return Classification
    End Function
#End Region

#Region "Constructors"
    Private Sub New(ByVal Name As String, ByVal USTAEquivalent As String)
        MarkAsChild()
        _Name = Name
        _USTAEquivalent = USTAEquivalent
    End Sub

    Private Sub New()
        'Prevent direction creation
        MarkAsChild()
    End Sub
#End Region

#Region "Data Access"
    Private Sub Fetch(ByVal dr As SafeDataReader)
        _ID = dr.GetInt32()
        _Name = dr.GetString()
        _USTAEquivalent = dr.GetString()
        MarkOld()
    End Sub

    Friend Sub Update()
        If Me.IsDeleted Then
            If Not Me.IsNew Then
                DataAccess.ExecUpdate("csla_delete_classification", ID)
            End If
            MarkNew()
        Else
            DataAccess.ExecUpdate("csla_save_classification", IDParam(_ID, IsNew), _Name, _USTAEquivalent)
            MarkOld()
        End If
    End Sub
#End Region

End Class
